### PR TITLE
Add connect/disconnect activity logging

### PR DESCRIPTION
**Pull request by iamigo**

### CODE REVIEW

File: src/util/emitUtils.js [hunk 1]
```diff
@@ -355,20 +359,44 @@ function initializeBotNamespace(inst, io) {
...
+        userInfo.name = responses[1].body.name;
+        userInfo.tokenName = responses[1].body.name;
+      }
+
+      if (toggle.isFeatureEnabled('activityLoggingUser')) {
+        logger.info(`activity=user:connect ipAddress=${userInfo.ipAddress} ` +
+          `nsp=${socket.nsp.name} process=${processName} ` +
+          `token=${userInfo.tokenName} user=${userInfo.name}`);
+      }
+
```
Comment (jgraff2) [inline src/util/emitUtils.js]: Could you move this into trackeConnectedRooms or some other function so it doesn't clutter up initializeNamespace?

File: src/util/emitUtils.js [hunk 2]
```diff
@@ -355,20 +359,44 @@ function initializeBotNamespace(inst, io) {
...
       addToRoom(socket);
-      trackConnectedRooms(socket);
+      trackConnectedRooms(socket, userInfo, processName);
       socket.emit('authenticated');
     })
     .catch((err) => {
       pubSubStats.trackAuthError();
+      logger.error('auth error', err.message);
```
Comment (jgraff2) [inline src/util/emitUtils.js]: We had a discussion about this in the token auth PR, with you arguing against logging auth errors. Why did you change your mind?

File: src/util/emitUtils.js [hunk 3]
```diff
@@ -355,20 +359,44 @@ function initializeBotNamespace(inst, io) {
...
   io.of(namespace).on('connect', (socket) =>
     Promise.join(
       validateIp(socket),
       validateTokenNewFormat(socket),
     )
-    .then(() => {
+    .then((responses) => {
+      const userInfo = {
+        ipAddress: responses[0],
+      };
```
Comment (jgraff2) [inline src/util/emitUtils.js]: Would be clearer with
.then(([ ipAddress ]) => {

File: src/util/emitUtils.js (hunk 2 above)
Comment (iamigo) [reply src/util/emitUtils.js]: You are absolutely right. Now that we're looking at actual real-time traffic, I find I do want to see a little more detail on the auth errors... mostly to make sure they *are* indeed expected.

File: src/util/emitUtils.js (hunk 1 above)
Comment (jgraff2) [reply src/util/emitUtils.js]: @iamigo

File: src/util/emitUtils.js [hunk 4]
```diff
@@ -355,23 +359,33 @@ function initializeBotNamespace(inst, io) {
...
-      addToRoom(socket);
-      trackConnectedRooms(socket);
-      socket.emit('authenticated');
-    })
-    .catch((err) => {
-      pubSubStats.trackAuthError();
-      socket.emit('auth error', err.message);
-      socket.disconnect();
-    })
+    Promise.join(validateIp(socket), validateTokenNewFormat(socket))
```
Comment (jgraff2) [inline src/util/emitUtils.js]: formatting

File: src/util/emitUtils.js [hunk 5]
```diff
@@ -423,24 +457,36 @@ function validateTokenOldFormat(socket) {
...
-                  .set('Authorization', token);
-  })
-  .timeout(conf.authTimeout);
+    .then((token) => Promise.join(
+      token,
+      jwtVerifyAsync(token, conf.secret),
+    ))
+    .then(([token, {username, tokenname}]) => {
+      const path = (username === tokenname) ? `v1/users/${username}`
+        : `v1/users/${username}/tokens/${tokenname}`;
```
Comment (jgraff2) [inline src/util/emitUtils.js]: formatting

File: src/util/emitUtils.js [hunk 6]
```diff
@@ -423,24 +457,36 @@ function validateTokenOldFormat(socket) {
...
+      return request.get(`${conf.apiUrl}/${path}`)
+        .set('Authorization', token);
```
Comment (jgraff2) [inline src/util/emitUtils.js]: formatting

File: src/util/emitUtils.js [hunk 7]
```diff
@@ -273,17 +277,17 @@ function initializePerspectiveNamespace(inst, io) {
...
-      socket.on('disconnect', () => {
-        pubSubStats.trackDisconnect();
-      });
-    })
-    .catch((err) => {
-      pubSubStats.trackAuthError();
-      socket.emit('auth error', err.message);
-      socket.disconnect();
-    })
+      .then(() => {
```
Comment (jgraff2) [inline src/util/emitUtils.js]: formatting

File: src/util/emitUtils.js [hunk 8]
```diff
@@ -334,17 +338,17 @@ function initializeBotNamespace(inst, io) {
...
```
Comment (jgraff2) [inline src/util/emitUtils.js]: formatting

File: src/util/emitUtils.js (hunk 7 above)
Comment (iamigo) [reply src/util/emitUtils.js]: I'm just accepting the formatting changes from the ide... we don't have linter on this project... I'll add one in another PR and reformat per style guide
